### PR TITLE
[Bugfix:TAGrading] handle null autograding

### PR DESCRIPTION
**Pull request by jchulton**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Several edge cases are generating frog robots due to nulls not properly being checked

### What is the new behavior?
null checks have been added, fixing the frog robots

### CODE REVIEW

File: site/app/views/submission/HomeworkView.php [hunk 1]
```diff
@@ -371,23 +371,25 @@ private function renderSubmitBox(Gradeable $gradeable, $graded_gradeable, $versi
...
-                $json = json_decode(file_get_contents($path), true);
-                if (!is_null($json)) {
-                    if (isset($json["git_user_id"])) {
-                        $github_user_id = $json["git_user_id"];
-                    }
-                    if (isset($json["git_repo_id"])) {
-                        $github_repo_id = $json["git_repo_id"];
+            if ($graded_gradeable !== null) {
+                // Get path to VCS_CHECKOUT
+                $gradeable_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "submissions", $gradeable->getId());
```
Comment (emaicus) [inline site/app/views/submission/HomeworkView.php:376]: Github won't let me comment on non-PR lines, so I'll put this comment here. 
* Line 322 looks as though it may still crash if `graded_gradeable` is null.
* Same for 478.

File: site/app/models/gradeable/AutoGradedVersion.php [hunk 2]
```diff
@@ -243,26 +243,28 @@ private function loadTestcases() {
...
-                    $results_public_path,
-                    $result_details['testcases'][$testcase->getIndex()]
-                );
-                $this->graded_testcases[$testcase->getIndex()] = $graded_testcase;
-                if (in_array($testcase, $config->getEarlySubmissionTestCases())) {
-                    $this->early_incentive_points += $graded_testcase->getPoints();
+        if(!is_null($config)) {
+            foreach ($config->getTestcases() as $testcase) {
+                if (!isset($result_details['testcases'][$testcase->getIndex()])) {
+                    // TODO: Autograding results file was incomplete.  This is a big problem, but how should
```
Comment (emaicus) [inline site/app/models/gradeable/AutoGradedVersion.php:249]: I agree that this fix will stop the bleeding, but I also want to know why a config would be null.

**Review by emaicus — CHANGES_REQUESTED** [review]
This PR should address robots 2, 3, and 4.

**Comment by emaicus** [comment]
I am making my suggested changes now.

File: site/app/models/gradeable/AutoGradedVersion.php (hunk 2 above)
Comment (jchulton) [reply site/app/models/gradeable/AutoGradedVersion.php:249]: My guess is loadTestCases can be called even when there is no autograding.

File: site/app/views/submission/HomeworkView.php (hunk 1 above)
Comment (jchulton) [reply site/app/views/submission/HomeworkView.php:378]: I've added null checks on graded_gradeable for these two lines

**Review by emaicus — APPROVED** [review]
Looks good!

**Comment by bmcutler** [comment]
@jchulton @emaicus 
please fix lint asap
